posts/2020-03-15-docs-rs-opt-into-fewer-targets.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: "docs.rs now allows you to choose your build targets"
+author: Joshua Nelson
+team: the docs.rs team <https://www.rust-lang.org/governance/teams/dev-tools#docs-rs>
+---
+
+Recently, [docs.rs] added a feature that allows crates to opt-out of building on all targets.
+If you don't need to build on all targets, you can enable this feature to reduce your build times.
+
+## What does the feature do?
+
+By default, docs.rs builds all crates published to [crates.io] for [every tier one target][metadata].
+However, most crates have the same content on all target.
+Of the platform-dependent crates, almost all target a single platform,
+and do not need to be built on other targets.
+For example, [`winapi`] only has documentation on the `x86_64-pc-windows-msvc`
+and `i686-pc-windows-msvc` targets, and is blank on all others.
+
+This feature allows you to request building only on specific targets.
+For example, [`winapi`] could opt into only building windows targets
+by putting the following in its `Cargo.toml`:
+
+```toml
+[package.metadata.docs.rs]
+# This also sets the default target to `x86_64-pc-windows-msvc`
+targets = ["x86_64-pc-windows-msvc", "i686-pc-windows-msvc"]
+```
+
+If you only need a single target, it's even simpler:
+
+```toml
+[package.metadata.docs.rs]
+# This builds only the default target (usually `x86_64-unknown-linux-gnu`)
+targets = []
+```
+
+See the [docs.rs documentation][metadata] for more details about how to opt-in.
+
+## How does this help my crate?
+
+Instead of building for every tier-one target, you can build for only a single target,
+reducing your documentation times by a factor of 6.
+This can especially help large crates or projects with many crates
+that take several hours to document.
+
+## How does this help docs.rs?
+
+Building all crates from crates.io can take a long time!
+Building fewer targets will allow us to reduce wait times for every crate.
+Additionally, this will decrease our fixed storage costs, allowing docs.rs to be sustainable into the future.
```

Comment:
Reading this, it seems like docs.rs is not sustainable atm and this change is required for it to be sustainable.

```suggestion
Additionally, this will decrease the growth of our storage costs, improving the sustainability of the project.
```